plugins/modules/site_workflow_manager.py

```diff
@@ -698,19 +893,34 @@ def site_requires_update(self):
             specified parameters, such as the site type and site details.
         """
 
-        type = self.have['current_site']['type']
-        updated_site = self.have['current_site']['site'][type]
-        requested_site = self.want['site_params']['site'][type]
+        if self.get_ccc_version_as_integer() <= self.version_2_3_5_3:
```

Comment:
```
   if self.get_ccc_version_as_integer() <= self.version_2_3_5_3:
        updated_site = current_site.get('site', {}).get(site_type)
        requested_site = self.want['site_params']['site'].get(site_type)
    else:
        self.log("Current site details: {}".format(current_site), "INFO")
        updated_site = current_site
        requested_site = self.want.get('site_params', {}).get('site', {}).get(site_type)

```

Reply:
site_type = self.have.get('current_site', {}).get('type')
In this line of code, site_type is being assigned the value of the type attribute from the current_site dictionary stored in self.have. 
so, using this line also